test/Jsr292/src/com/ibm/j9/jsr292/indyn/IndyTest.java

```diff
@@ -333,28 +333,15 @@ public void test_indyn_ldc_BootstrapMethod_type() {
 		}
 	}
 	
-	//Negative test : Creating an invalid MethodType using LDC
+	//Negative test : Creating a MethodType with a non-existent class using LDC
 	@Test(groups = { "level.extended" })
-	public void test_indyn_ldc_invalid_type() {
-		
-		boolean typeNotPresentExceptionThrown = false;
-		
+	public void test_indyn_ldc_non_existent_class_methodtype() {
 		try {
 			MethodType mtToString = com.ibm.j9.jsr292.indyn.GenIndyn.ldc_invalid_type();
-		} catch ( java.lang.TypeNotPresentException e ) {
-			typeNotPresentExceptionThrown = true;
-		}
-		
-		//Call it again
-		try {
-			MethodType mtToString = com.ibm.j9.jsr292.indyn.GenIndyn.ldc_invalid_type();
-		} catch ( java.lang.TypeNotPresentException e ) {
-			typeNotPresentExceptionThrown = true;
-		}
-		
-		if ( typeNotPresentExceptionThrown == false ) {
-			Assert.fail("TypeNotPresentException not thrown when invalid type descriptor is used to create MethodType via LDC");
+		} catch (NoClassDefFoundError e) {
+			return;
 		}
+		Assert.fail("NoClassDefFoundError not thrown when non-existent class is used to create MethodType via LDC");
```

Comment:
This `Assert.fail` can be moved after `MethodType mtToString = com.ibm.j9.jsr292.indyn.GenIndyn.ldc_invalid_type();`, and `return` within `catch` block can be removed.

Reply:
fixed.